content/updates/terraform-null-label.md

```diff
@@ -0,0 +1,200 @@
+    subnets            = ["subnet-abc12345", "subnet-xyz67890"]
+
+    # terraform-null-label outputs the merged tags input and all ID element inputs as a map value
+    # i.e. { namespace: "mp", stage: "prod", Team: "Reporting", ManagedByTerraform: "True", ... }
+    tags               = module.public_alb_label.tags
+  }
+
+  resource "aws_security_group" "alb_sg" {
+    name        = module.public_alb_label.id
+    description = "Allow inbound traffic over HTTPS"
+    vpc_id      = "vpc-12345678"
+    tags        = module.public_alb_label.tags
+
+    ingress {
+      from_port   = 443
+      to_port     = 443
+      protocol    = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
+    }
+  }
+```
+
+It’s worth noting that in our examples, we’re using the AWS provider, but the label module can be used anywhere in Terraform regardless of provider or use-case.
+
+In the above example, the `terraform-null-label` module lets engineers define some common-sense values for the ID elements like stage and namespace, and then it can be used to apply a consistent name and tagging scheme to every resource and/or module that the team uses. "Write once, read many" and DRY (Don’t repeat yourself) are important factors we talk a lot about within good application code, and null-label enables us to easily incorporate those ideas into our platform code.
+
+Now, obviously that’s a very simplified and abstract example. In complex configurations, there are likely going to be many resources and modules with different usage, and in this case even with `terraform-null-label` it can start to get cumbersome to have to define new label modules each time you need to have variations in your ID elements. There is a solution to that however as one of the module’s most powerful features addresses exactly that: the `context` input + output value.
+
+Going back to our example, what if we needed to deploy another internal load balancer alongside the public one? That is going to share a lot of the same label id elements with a couple of overrides. Here’s how easy that is:
+
+```hcl
+  module "public_alb_label" {
+    source  = "cloudposse/label/null"
+    version = "0.25.0"
+
+    namespace   = "mp"
+    stage       = "prod"
+    environment = "uw2"
+    name        = "alb"
+    attributes  = ["public"]
+
+    tags = {
+      BusinessUnit       = "ENG"
+      Team               = "Reporting"
+      ManagedByTerraform = "True"
+    }
+  }
+
+  module "private_alb_label" {
+    source  = "cloudposse/label/null"
+    version = "0.25.0"
+
+    attributes  = ["private"]
+
+    # Here's the important bit
+    context = module.public_alb_label.context
+  }
+```
+
+By passing in the context output from the first label module, all of the original values are present, with only the defined ones overridden. With that, you get:
+
+* `module.public_alb_label.id` equates to `mp-uw2-prod-alb-public`
+* `module.public_alb_label.tags` equates to:
+
+   ```json
+   {
+     Namespace: "mp",
```

Comment:
These blobs of data aren't valid JSON, so the syntax highlighter is showing this weirdly in the rendered page. What language was this meant to be?

Reply:
I dunno, I was trying to show them as JSON. I'll update it to be less JSON-y (JSON sucks).